src/cascadia/TerminalSettingsEditor/GlobalAppearanceViewModel.cpp

```diff
@@ -260,4 +260,13 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     {
         _GlobalSettings.DisableAnimations(!value);
     }
+
+    void GlobalAppearanceViewModel::ShowTitlebarToggled(const winrt::Windows::Foundation::IInspectable& /* sender */, const RoutedEventArgs& /* args */)
+    {
+        // Set AlwaysShowTabs to true if ShowTabsInTitlebar was toggled OFF -> ON

```

Comment:
Technically this happens because ShowTabsInTitlebarToggled fires **before** the value of ShowTabsInTitlebar has changed; this is safe, but it may be worth calling out. This is an interesting part of XAML data binding that I did not expect!